### PR TITLE
configmap pkg: Update method added

### PR DESCRIPTION
**Pull request by elenagerman**
configmap pkg: Update method added

### CODE REVIEW

**Comment by elenagerman** [comment]
@trewest @kononovn @achuzhoy @cdvultur PTAL

**Comment by elenagerman** [comment]
@trewest @kononovn @achuzhoy @cdvultur PTAL

File: pkg/configmap/configmap_test.go [hunk 1]
```diff
@@ -196,20 +195,34 @@ func TestDelete(t *testing.T) {
 	}
 }
 
-func TestWithOptions(t *testing.T) {
```
Comment (trewest) [inline pkg/configmap/configmap_test.go:199]: Seems you removed the withoptions test?

File: pkg/secret/secret.go [hunk 2]
```diff
@@ -33,6 +33,12 @@ func NewBuilder(apiClient *clients.Settings, name, nsname string, secretType cor
 		"Initializing new secret structure with the following params: %s, %s, %s",
```
Comment (trewest) [inline pkg/secret/secret.go:33]: Please move secret updates to a separate PR

Comment (elenagerman) [reply pkg/secret/secret.go]: yea,it had to be a separate mr, it's by mistake here. thanks!

File: pkg/configmap/configmap_test.go [hunk 3]
```diff
@@ -196,20 +196,34 @@ func TestDelete(t *testing.T) {
...
-		return builder, errors.New("error")
-	})
+	for _, testCase := range testCases {
+		testCase.testCM.WithData(testCase.testData)
+		_, err := testCase.testCM.Update()
 
-	assert.Equal(t, "error", testBuilder.errorMsg)
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
```
Comment (trewest) [inline pkg/configmap/configmap_test.go:223]: Can you change expectedError to an error type using fmt.Errorf and always check it (even if it is nil)?
```
assert.Equal(t, testCase.expectedError, err)
```
Then just check the data if expectedError is nil
```
if testCase.expectedError == nil {
  assert.Equal(t, testCase.testData, testCase.testCM.Definition.Data)
}

Comment (elenagerman) [reply pkg/configmap/configmap_test.go:223]: @trewest I cannot use err nil verification in this case, cause this object is not really exists:
testCM:        buildTestBuilderWithFakeObjects([]runtime.Object{})

If you insist such verification has to be added I'll need to do a full unit-test refactoring

Comment (trewest) [reply pkg/configmap/configmap_test.go:223]: Yes, please make sure that the resource actually exists so we can test it thoroughly. Please look at the deployment test as an example - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment_test.go#L492

File: pkg/configmap/configmap_test.go [hunk 4]
```diff
@@ -196,20 +197,60 @@ func TestDelete(t *testing.T) {
...
+
+		// Set a value in the definition to test the update
+		testBuilder.Definition.Data = map[string]string{"key1": "value1", "key2": "value2"}
+
+		// Perform the update
+		result, err := testBuilder.Update()
+
+		// Assert the result
+		assert.NotNil(t, testBuilder.Definition)
+
```
Comment (trewest) [inline pkg/configmap/configmap_test.go:245]: Looks like you didn't use the expectedError parameter. Please just check assert.Equal(t, testCase.expectedError, err)

Comment (elenagerman) [reply pkg/configmap/configmap_test.go:241]: in provide by you example there is no such verification; I just remove expected error var at all

**Review by trewest — DISMISSED** [review]
/lgtm

File: pkg/configmap/configmap.go [hunk 5]
```diff
@@ -157,6 +157,32 @@ func (builder *Builder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// Update renovates the existing kedaController object with kedaController definition in builder.
```
Comment (kononovn) [inline pkg/configmap/configmap.go:160]: Please fix the comment.

**Review by kononovn — CHANGES_REQUESTED** [review]
1 small comment to fix.

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]
/lgtm
